project_generator/exporters/uvision.py

```diff
@@ -146,12 +147,16 @@ def fix_paths(self, data, rel_path):
             fixed_paths.append(join(rel_path, normpath(path)))
         data['includes'] = fixed_paths
         fixed_paths = []
-        for path in data['source_files_lib']:
-            fixed_paths.append(join(rel_path, normpath(path)))
-        data['source_files_lib'] = fixed_paths
-        fixed_paths = []
-        for path in data['source_files_obj']:
-            fixed_paths.append(join(rel_path, normpath(path)))
+        for k in data['source_files_lib'][0].keys():
```

Comment:
this also expects as sources a dictionary, which is not valid always. If a project is just flat (no groups), this generates attribute error.